week7/answers.sql

```diff
@@ -1,3 +1,256 @@
+    i.armor,
+    i.damage
+FROM 
+    characters AS c
+INNER JOIN 
+    equipped AS e ON c.character_id = e.character_id
+INNER JOIN 
+    items AS i ON e.item_id = i.item_id -- this is where it should allow the view
+ORDER BY item_name;
+
+
+-- basically rinse repeat but for all of them in a team
+CREATE VIEW team_items AS
+SELECT 
+    t.team_id,
+    t.name AS team_name,
+    i.name AS item_name,
+    i.armor,
+    i.damage
+FROM 
+    teams t
+INNER JOIN 
+    team_members tm ON t.team_id = tm.team_id -- didn't know what else to call tm
+INNER JOIN 
+    characters c ON tm.character_id = c.character_id
+INNER JOIN 
+    inventory inv ON c.character_id = inv.character_id
+INNER JOIN 
+    items i ON inv.item_id = i.item_id
+UNION
+SELECT 
+    t.team_id,
+    t.name AS team_name,
+    i.name AS item_name,
+    i.armor,
+    i.damage
+FROM 
+    teams t
+INNER JOIN 
+    team_members tm ON t.team_id = tm.team_id
+INNER JOIN 
+    characters c ON tm.character_id = c.character_id
+INNER JOIN 
+    equipped e ON c.character_id = e.character_id
+INNER JOIN 
+    items i ON e.item_id = i.item_id;
+
+DELIMITER ;; -- I didn't know these delimeters could be set to be anything :-)
+CREATE FUNCTION armor_total(character_id INT) RETURNS INT
+DETERMINISTIC -- UPDATE - Github complained about this missing 
+BEGIN
+    DECLARE total_armor INT; -- I still struggle with when is it that vars need an @ symbol 
+			     -- before the declaration... I think they dont when they're local
+
+    SELECT SUM(i.armor) INTO total_armor
+    FROM equipped e
+    INNER JOIN items i ON e.item_id = i.item_id
+    WHERE e.character_id = character_id;
+
+    RETURN total_armor;
+END ;;
+DELIMITER ;
+
+DELIMITER ;; 
+CREATE PROCEDURE attack(IN id_of_character_being_attacked INT, IN id_of_equipped_item_used_for_attack INT)
+BEGIN
+    DECLARE armor INT;
+    DECLARE damage INT;
+    DECLARE health INT;
+
+    -- armor of character being attacked
+    SET armor = armor_total(id_of_character_being_attacked);
+
+    -- damage of item used to attack
+    SELECT damage INTO damage FROM items WHERE item_id = id_of_equipped_item_used_for_attack;
+
+    -- calc the damage
+    SET damage = damage - armor; -- substracting armor from damage
+    IF damage < 0 THEN
+        SET damage = 0; -- this is assuming we don't care about the durability of the armor
+    END IF;				      -- or it regenerates on it's onw like in Call of Duty	
+
+    -- set the health
+    SELECT health INTO health FROM character_stats WHERE character_id = id_of_character_being_attacked;
+
+    -- Sub the damage
+    SET health = health - damage; -- I'm comparing health with the damage from before
+
+	-- now we update the health or delete the poor victim of database violence 
+    IF health > 0 THEN
+        UPDATE character_stats SET health = health WHERE character_id = id_of_character_being_attacked;
+    ELSE
+        DELETE FROM character_stats WHERE character_id = id_of_character_being_attacked;
```

Comment:
You only need one DELETE if you set your ON DELETE clauses in your FOREIGN KEY constraints (Week 4 & 5: Foreign Keys).